lib/cache/BaseCache.ts

```diff
@@ -16,6 +16,7 @@ export class BaseCache<T extends CacheEntry> {
   private readonly CACHE_MAX_AGE_MS = 7 * 24 * 60 * 60 * 1000; // 1 week in milliseconds
   private readonly CLEANUP_PROBABILITY = 0.01; // 1% chance
 
+  protected enableCaching: boolean;
```

Comment:
It doesn't make sense to add this field here. Rather, we should just not init a cache if `enableCaching` is set to false in `LLMProvider` and `actHandler`. It makes more sense to make the cache default `undefined` in `LLMProvider` and `actHandler`, and before a cache operation there, check `enableCaching`.